src/v/cluster/topics_frontend.cc

```diff
@@ -1284,4 +1284,47 @@ ss::future<std::error_code> topics_frontend::decrease_replication_factor(
       _stm, _features, _as, std::move(cmd), timeout);
 }
 
+allocation_request make_allocation_request(
+  model::partition_id pid,
+  topic_configuration topic_cfg,
+  const std::vector<model::node_id>& new_replicas) {
+    allocation_request req(get_allocation_domain(topic_cfg.tp_ns));
+    req.partitions.reserve(1);
+    allocation_constraints constraints;
+    constraints.hard_constraints.push_back(
+      ss::make_lw_shared<hard_constraint_evaluator>(on_nodes(new_replicas)));
+    req.partitions.emplace_back(
+      pid, topic_cfg.replication_factor, std::move(constraints));
+    return req;
+}
+
+ss::future<result<std::vector<partition_assignment>>>
+topics_frontend::generate_reassignments(
+  model::partition_id pid,
+  model::topic_namespace topic_ns,
+  const std::vector<model::node_id> new_replicas) {
+    auto topic_cfg = _topics.local().get_topic_cfg(topic_ns);
```

Comment:
please use `get_replication_factor` the replication factor field in topic configuration is deprecated.